### PR TITLE
introduced twist_mux and new velocity smoother

### PR DESCRIPTION
**Pull request by benmaidel**
- added twist_mux launch files and configs as symlinks (common folder)
- added new velocity_smoother launch files and configs
- adjusted topic names to new structure

coupled with:
- https://github.com/ipa320/ipa_navigation/pull/87
- https://github.com/ipa320/cob_navigation/pull/57
- https://github.com/ipa320/cob_command_tools/pull/125
- https://github.com/ipa320/cob_control/pull/67

successfully tested on raw3-3

### CODE REVIEW

File: cob_bringup/controllers/base_controller.launch [hunk 1]
```diff
@@ -3,23 +3,32 @@
...
+			<arg name="robot" value="$(arg robot)" />
+			<arg name="cmd_vel_out" default="velocity_smoother/command"/>
+			<arg name="marker_out" default="twist_mux/marker"/>
+		</include>
+
+		<!-- velocity smoother -->
+		<include file="$(find cob_bringup)/tools/velocity_smoother.launch" >
+			<arg name="robot" value="$(arg robot)" />
+  		<arg name="config_file"           default="$(arg pkg_hardware_config)/$(arg robot)/config/velocity_smoother.yaml"/>
+  		<arg name="raw_cmd_vel_topic"     default="velocity_smoother/command"/>
```
Comment (fmessmer) [inline cob_bringup/controllers/base_controller.launch:27]: indentation

**Comment by fmessmer** [comment]
FR ok...looks very consistent!
Just some minor indentation issues...but we can fix that directly after the merge

File: cob_bringup/controllers/base_controller.launch [hunk 2]
```diff
@@ -3,23 +3,32 @@
...
+  		<arg name="smooth_cmd_vel_topic"  default="twist_controller/command"/>
+  		<arg name="robot_cmd_vel_topic"   default="twist_controller/command"/>
```
Comment (mgruhler) [inline cob_bringup/controllers/base_controller.launch:29]: If those two should always be the same (which they should, afaik), wouldn't it make sense to "link" the argument from one to the other?
e.g. `<arg name="robot_cmd_vel_topic"   default="$(arg smoot_cmd_vel_topic)"/>`
Or is there a case where this is wrong?

File: cob_bringup/controllers/base_controller_plugin.launch [hunk 3]
```diff
@@ -13,6 +13,23 @@
...
+    <include file="$(find cob_bringup)/tools/velocity_smoother.launch" >
+      <arg name="robot" value="$(arg robot)" />
+      <arg name="config_file"           default="$(arg pkg_hardware_config)/$(arg robot)/config/velocity_smoother.yaml"/>
+      <arg name="raw_cmd_vel_topic"     default="velocity_smoother/command"/>
+      <arg name="smooth_cmd_vel_topic"  default="twist_controller/command"/>
+      <arg name="robot_cmd_vel_topic"   default="twist_controller/command"/>
+      <arg name="odom_topic"            default="odom_controller/odometry"/>
+    </include>
+
 	</group>
```
Comment (mgruhler) [inline cob_bringup/controllers/base_controller_plugin.launch:33]: indentation as well

File: cob_bringup/controllers/base_controller.launch (hunk 2 above)
Comment (benmaidel) [inline cob_bringup/controllers/base_controller.launch:29]: nope, they should not always be the same. `robot_cmd_vel_topic` is meant to be the actual robot velocity and `smooth_cmd_vel_topic` the desired robot velocity. In our case, we should use the odometry topic to compare actual and desired velocity. Look here: https://github.com/ipa320/cob_robots/pull/377/files#diff-82d0476ef449ef932585fb58723ee231R23

Unfortunately I have not yet tested the smoother with odometry feedback

Comment (mgruhler) [inline cob_bringup/controllers/base_controller.launch:29]: understood, thanks

**Comment by NadiaHG** [comment]
FR: OK
Should be tested on cob4

**Pull request merged**